libraries/net/eth/EthernetInterface/EthernetInterface.h

```diff
@@ -45,7 +45,7 @@ class EthernetInterface {
   * \param gateway the gateway to use
   * \return 0 on success, a negative number on failure
   */
-  static int init(const char* ip, const char* mask, const char* gateway);
+  static int init(const char* ip, const char* mask, const char* gateway, const char * dns = NULL);
```

Comment:
Well, normally lwip gets its dns servers from the dhcp request, so if this class allows static addresses, the dns server does need to be provided manually.

This style does match the existing style, though may prove to not be scalable.